packages/wallets/src/features/cashier/modules/TransactionStatus/__tests__/TransactionStatus.spec.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { useActiveWalletAccount, useCryptoTransactions } from '@deriv/api';
+import { fireEvent, render, screen } from '@testing-library/react';
+import TransactionStatus from '../TransactionStatus';
+
+jest.mock('@deriv/api', () => ({
+    useActiveWalletAccount: jest.fn(() => ({
+        data: null,
+        error: null,
+        isLoading: false,
+        refetch: jest.fn(),
+    })),
+    useCryptoTransactions: jest.fn(() => ({
+        data: null,
+        error: null,
+        isLoading: false,
+        resetData: jest.fn(),
+        subscribe: jest.fn(),
+        unsubscribe: jest.fn(),
+    })),
+}));
+
+describe('TransactionStatus component', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should render default component correctly', async () => {
+        render(<TransactionStatus />);
+
+        expect(screen.getByText('Transaction status')).toBeInTheDocument();
+        expect(screen.queryByTestId('dt_wallets_loader')).not.toBeInTheDocument();
+        expect(screen.queryByRole('button', { name: 'Refresh' })).not.toBeInTheDocument();
+    });
+
+    it('should render success state correctly when wallet data exists', async () => {
+        (useActiveWalletAccount as jest.Mock).mockImplementation(() => ({
+            data: [],
+            error: null,
+            isLoading: false,
+            refetch: jest.fn(),
+        }));
+        render(<TransactionStatus />);
+
+        expect(screen.getByText('Transaction status')).toBeInTheDocument();
+        expect(screen.getByText('No recent transactions.')).toBeInTheDocument();
+    });
+
+    it('should render loading state correctly for useActiveWalletAccount', async () => {
+        (useActiveWalletAccount as jest.Mock).mockImplementation(() => ({
+            data: null,
+            error: null,
+            isLoading: true,
+            refetch: jest.fn(),
+        }));
+
+        render(<TransactionStatus />);
+        expect(screen.getByText('Transaction status')).toBeInTheDocument();
+        expect(screen.getByTestId('dt_wallets_loader')).toBeInTheDocument();
+    });
+
+    it('should render loading state correctly for useCryptoTransactions', async () => {
```

Comment:
thumbs up for testing separately both useActiveWallet loading and useCryptoTransactions :D